[New] `jsx-indent-props`: add `ignoreTernaryOperator` option

SebastianZimmer opened the pull request.

ljharb reviewed (changes requested):
Let's pick an option name where absence and `false` are the same.

How about, `ignoreTernaryOperator` that defaults to false?

SebastianZimmer commented:
> Let's pick an option name where absence and false are the same.
> 
> How about, ignoreTernaryOperator that defaults to false?

Good idea, I've changed it that way and adjusted documentation and tests.
